src/IIIFingest/asset.py

```diff
@@ -65,20 +74,32 @@ def upload(
         self, bucket_name: str = "", s3_path: Optional[str] = None, boto_session=None
     ) -> str:
         """
-        Uploads the asset to the designated bucket.
+        Uploads the asset to the designated bucket. Chooses a strategy based on
+        whether the asset has a filepath or a fileobj.
         """
-        self.s3key = upload_image_get_metadata(
-            image_path=self.filepath,
-            bucket_name=bucket_name,
-            s3_path=s3_path,
-            session=boto_session,
-        )
+        if self.filepath:
```

Comment:
Are we considering the `filepath` and `fileobj` attributes to be mutually exclusive? I doubt a user would try to construct an `Asset` with both, but if they did, we might want to clarify whether we're going to privilege the `filepath` over the `fileobj` (maybe note that in the class docstring) or simply raise an error in the constructor and require one or the other.

Reply:
Right now the two ways that assets get created are the `from_file` and `from_fileobj` methods, which create assets with only one or the other, but putting a note in the docstring for the `Asset` class that `Asset` creation is expected to go through one of those classes would be better.

Reply:
Yeah it wouldn't hurt to make the intention clear with a note on the usage. I think the only reason to add error handling to the `Asset` constructor would be as a fail-safe in case someone wasn't using one of those prescribed methods. But a note would suffice imo!